pkg/operator/configobservation/apiserver/observe_apiserver.go

```diff
@@ -116,9 +116,8 @@ func observeDefaultUserServingCertificate(apiServer *configv1.APIServer, recorde
 	return observedConfig, syncActionRules{"user-serving-cert": servingCertSecretName}, errs
 }
 
-// observeNamedCertificates observes user managed Secrets containing TLS cert info for serving secure traffic to
-// specific hostnames.
-func observeNamedCertificates(apiServer *configv1.APIServer, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, syncActionRules, []error) {
+// observeNamedCertificates observes Secrets containing TLS cert info for serving secure traffic to specific hostnames.
+func observeNamedCertificates(listers configobservation.Listers, apiServer *configv1.APIServer, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, syncActionRules, []error) {
```

Comment:
you forgot to use the dns lister

Reply:
I didn't use it on purpose. I needed it for the `baseDomain` part of `<clusterName>-api.<baseDomain>` hostname, but I do not know where to get the `clusterName` from. I left it in for your use when you figure out how to get the `clusterName`.